__tests__/app.test.js

```diff
@@ -0,0 +1,53 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const data = require("../db/data/test-data");
+const endpoints = require("../endpoints.json");
+
+beforeAll(() => seed(data));
+afterAll(() => db.end());
+
+describe("Topics API", () => {
+	describe("Get Request", () => {
+		test("Get: 200, should return all topics", () => {
+			return request(app)
+				.get("/api/topics")
+				.expect(200)
+				.then(({ body }) => {
+					const { topics } = body;
+					expect(topics).toBeInstanceOf(Array);
+					expect(topics.length).not.toBeLessThan(1);
+					topics.forEach((topic) => {
+						expect(Object.keys(topic)).toEqual(["slug", "description"]);
+					});
+				});
+		});
+	});
+});
+describe("Endpoints API", () => {
+	describe("Get Request", () => {
+		test("endpoints.json formatting correct", () => {
+			expect(endpoints).toBeInstanceOf(Object);
+		});
+		test("Get: 200, should return a list of endpoints from json file", () => {
+			return request(app)
+				.get("/api")
+				.expect(200)
+				.then(({ body }) => {
+					const endpointList = body.endpoints;
+					expect(endpointList).toEqual(endpoints);
+					expect(endpointList).not.toBe(endpoints);
+				});
+		});
+	});
+});
+
+// Implement tests later when more methods have been implemented
+describe("Error Testing", () => {
+	test("Get: 404, Returns error 404 when given an invalid url", () => {
+		return request(app).get("/api/topic").expect(404);
+	});
+	//test("GET: 400", () => {});
+	//test("should be able to handle a get request when there are no topics", () => {});
```

Comment:
Be sure to remove any commented out code before submitting work for review. A pull request should represent your best view of what you believe to be your work in its final state, so tidying up lines like these in advance of submission is something to do.